lib/import/mobility.js

```diff
@@ -0,0 +1,188 @@
+    csv
+    .fromStream(stream, {headers: true})
+    .on('data', function(data) {
+      if (verbose && count % 1000 === 0) {console.log(count);}
+      var o = {};
+      o.date = yyyymmdd_to_iso(data.DATE);
+      o.origin_country_code = country_code;
+      o.destination_country_code = country_code;
+      o.origin_admin_code = country_code + '-' + data.ORIGIN;
+      o.destination_admin_code = country_code + '-' + data.DESTINATION;
+      o.count = data.COUNT;
+      mobility_objects.push(new Mobility(o));
+      ++count;
+    })
+    .on('end', function() {
+      bulk_save(mobility_objects, verbose)
+      .then(function(err) {
+        if (err) {return reject(err);}
+        resolve();
+      });
+    });
+  });
+}
+
+/**
+ * Import mobility collection into mongo.
+ *
+ * @param{array} ary - array of mobility objects
+ * @param{bool} verbose - whether to log messages while importing.
+ * @return{promise} Fulfilled once all records are saved.
+ */
+function bulk_save(ary, verbose) {
+  var debug_msg = 'Have left to save';
+  return new Promise(function(resolve, reject) {
+  /**
+   * Save a splice of array of mobility records.
+   * @param{array} ary - array of mobility objects
+   */
+    function save(ary) {
+      if (ary.length > 0) {
+        if (verbose) { console.log(debug_msg, ary.length);}
+        Mobility.insertMany(ary.splice(0, 1000))
+        .then(function() {
+          save(ary);
+        })
+        .catch(function(err) {
+          return reject(err);
+        });
+      } else {
+        resolve();
+      }
+    }
+    save(ary);
+  });
+}
+
+/**
+ * Return list of mobility files in directory.
+ *
+ * @param{string} dir - directory where mobility files are stored
+ * @return{promise} Fulfilled once all files are in list
+ */
+function get_mobility_files(dir) {
+  return new Promise(function(resolve, reject) {
+    fs.readdir(dir, function(err, list) {
+      if (err) { console.log(err); return reject(err);}
+      list = list.filter(function(e) {
+        return e.match(/.csv/);
+      });
+      resolve(list);
+    });
+  });
+}
+
+/**
+ * Import mobility from CSV to Mongo.
+ *
+ * @param{string} dir - directory where mobility csv files exist.
+ * @param{string} country_code - lowercase ISO 3166-1 alpha-2 country code
+ *   https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+ * @param{bool} verbose - whether to log messages while importing.
+ * @return{promise} Fulfilled once all records are saved.
+ *
+ * Needs to be connected to mongo before calling.
+ */
+function import_mobility(dir, country_code, verbose) {
+  var all_done = [];
+  return get_mobility_files(dir)
+  .then(function(list) {
+    list.forEach(function(csv_file) {
+      var promise = new Promise(function(resolve) {
+        import_file(dir + '/' + csv_file, country_code, verbose)
+        .then(function() {
+          resolve();
+        });
+      });
+      all_done.push(promise);
+    });
+    return Promise.all(all_done);
+  });
+}
+
```

Comment:
@jetpack: I'm concerned that this function will overload mongo with concurrent writes. 

It begins by reading files in a directory, and then opening each and saving chunks of lines by the thousand. The original design, present in the old resources repo, opened one file at a time in sequence.